# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/TransportSecurityInfoUtils.cpp
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 1
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e ipc/glue/TransportSecurityInfoUtils.cpp
--- a/ipc/glue/TransportSecurityInfoUtils.cpp	Mon Oct 21 19:33:02 2019 +0000
+++ b/ipc/glue/TransportSecurityInfoUtils.cpp	Mon Oct 21 19:49:01 2019 +0000
@@ -76,37 +76,4 @@
   return true;
 }
 
-void ParamTraits<nsIX509CertList*>::Write(Message* aMsg,
-                                          nsIX509CertList* aParam) {
-  bool nonNull = !!aParam;
-  WriteParam(aMsg, nonNull);
-  if (!nonNull) {
-    return;
-  }
-
-  aParam->SerializeToIPC(aMsg);
-}
-
-bool ParamTraits<nsIX509CertList*>::Read(const Message* aMsg,
-                                         PickleIterator* aIter,
-                                         RefPtr<nsIX509CertList>* aResult) {
-  bool nonNull = false;
-  if (!ReadParam(aMsg, aIter, &nonNull)) {
-    return false;
-  }
-
-  if (!nonNull) {
-    *aResult = nullptr;
-    return true;
-  }
-
-  RefPtr<nsIX509CertList> certList = new nsNSSCertList();
-  if (!certList->DeserializeFromIPC(aMsg, aIter)) {
-    return false;
-  }
-
-  *aResult = certList.forget();
-  return true;
-}
-
 }  // namespace IPC